### DISCUSSION TITLE
Hardware Acceleration for Intel Cherryview/Braswell gpu

### DISCUSSION THREAD

**Discussion by ekt-** [blakeblackshear/frigate#3402] [Ask A Question]
I have frigate installed as an addon in Hassio (OS) on a mini pc based on
cpu: Intel Atom x5-Z8350 
gpu: Intel Cherryview/Braswell SoC - Integrated Graphics

frigate works great (awesome)
but if I try to enable hardware acceleration i get this error in the logs
`AVHWDeviceContext @ 0x56521abe6780] No VA display found for device /dev/dri/renderD128 `

these are the parameters I used
```
hwaccel_args:
  - -hwaccel
  - vaapi
  - -hwaccel_device
  - /dev/dri/renderD128
  - -hwaccel_output_format
  - yuv420p
```
I tried both the 'Frigate NVR' addon and the 'Full Access' version but the error message is the same (and if I get it right the 'protection mode' seems only to apply to rasppi)

do you expect acceleration to work on that gpu? is there any other thing I can do?

**Comment by NickM-27** [comment]
According to https://superuser.com/questions/1437496/hardware-accelerated-encoding-using-ffmpeg-on-intel-atom-based-platform-upboard you might want to try the qsv variant of hwaccel args

**Reply by ekt-** [reply]
thanks @NickM-27 

i've added this options
```
      hwaccel_args:
        - -hwaccel
        - qsv
        - -hwaccel_output_format
        - yuv420p
```
it seems to work, or at least I had no errors. I will monitor the CPU usage in the next few hours to see if it gets lower or not.

Not sure if that option alone is all it takes to activate qsv. looking at ffmpeg docs perhaps I need also
`ffmpeg -init_hw_device qsv=qsv:hw -filter_hw_device hw  -hwaccel qsv `

because **-init_hw_device** seems necessary to initialize a hardware device, while **-hwaccel** is specifically meant for decoding

**Reply by NickM-27** [reply]
Keep in mind that some of this is handled behind the scenes in frigate. Those init commands shouldn't be needed to be entered in your config (just like the docs don't recommend this).

**Reply by ekt-** [reply]
Ok. I can't see them in the 'ffmpeg_cmds' in the debug output and I assumed the 'cmd' string is what exactly is used to spawn the ffmpeg process

**Reply by NickM-27** [reply]
I'll have to see if I can find it in the ffmpeg docs, but the `--init...` is only needing to be sent to ffmpeg if it is not able to find the hwaccel device on its own. Assuming you passed it to the correct `/dev/dri/renderD128` then it is finding it automatically

**Reply by ekt-** [reply]
`-hwaccel_device` which I'm not passing right now, I'll test that too. docs says

> This option only makes sense when the -hwaccel option is also specified. It can either refer to an existing device created with -init_hw_device by name, or it can create a new device as if ‘-init_hw_device’ type:hwaccel_device were called immediately before.

**Reply by ekt-** [reply]
unfortunately `-hwaccel qsv` alone did not produce any appreciable improvement on the cpu. (in the below pic 30 june is without flag, 1 july is with the flag)
![image](https://user-images.githubusercontent.com/4489148/176934294-564567e1-1442-4757-ab4e-7eb02bb7cc79.png)

the curve shape shows that during the day there is much more activity than at night, hence the CPU works harder. if hardware acceleration works, I expect to see a cpu reduction especially at night, with quieter times, when no inferences occur, but only motion detection, so only cpu usage
